app/src/main/java/com/github/se/wanderpals/model/repository/TripsRepository.kt

```diff
@@ -256,6 +322,201 @@ class TripsRepository(
+   *   occurs. The method logs an error and returns `null` in case of failure.
+   */
+  suspend fun getExpenseFromTrip(tripId: String, expenseId: String): Expense? =
+      withContext(dispatcher) {
+        try {
+          val documentSnapshot =
+              tripsCollection
+                  .document(tripId)
+                  .collection(FirebaseCollections.TRIP_EXPENSES_SUBCOLLECTION.path)
+                  .document(expenseId)
+                  .get()
+                  .await()
+          val firestoreExpense = documentSnapshot.toObject<FirestoreExpense>()
+          if (firestoreExpense != null) {
+            firestoreExpense.toExpense()
+          } else {
+            Log.e(
+                "TripsRepository",
+                "getExpenseFromTrip: Not found Expense $expenseId from trip $tripId.")
+            null
+          }
+        } catch (e: Exception) {
+          Log.e(
+              "TripsRepository",
+              "getExpenseFromTrip: Error getting Expense $expenseId from trip $tripId.",
+              e)
+          null // error
+        }
+      }
+
+  /**
+   * Fetches all expenses associated with a specific trip. This method retrieves an array of
+   * expenses by leveraging the `getExpenseFromTrip` method for each expense ID found in the trip
+   * document.
+   *
+   * @param tripId The unique identifier of the trip for which expenses are being retrieved.
+   * @return A list of `Expense` objects. Returns an empty list if the trip is not found or if an
+   *   error occurs during fetching. Errors are logged and an empty list is returned in these cases.
+   */
+  suspend fun getAllExpensesFromTrip(tripId: String): List<Expense> =
+      withContext(dispatcher) {
+        try {
+          val trip = getTrip(tripId)
+
+          if (trip != null) {
+            val expenseIds = trip.expenses
+            expenseIds.mapNotNull { expenseId -> getExpenseFromTrip(tripId, expenseId) }
+          } else {
+            Log.e("TripsRepository", "getAllExpensesFromTrip: Trip not found with ID $tripId.")
+            emptyList()
+          }
+        } catch (e: Exception) {
+
+          Log.e(
+              "TripsRepository",
+              "getAllExpensesFromTrip: Error fetching Expenses to trip $tripId.",
+              e)
+          emptyList()
+        }
+      }
+
+  /**
+   * Adds a new expense to a specified trip. This method generates a unique identifier for the new
+   * expense, stores it in the Firestore, and updates the trip document to include this new expense
+   * ID.
+   *
+   * @param tripId The unique identifier of the trip to which the expense is being added.
+   * @param expense The `Expense` object to be added.
+   * @return The unique identifier of the newly added expense if the operation is successful, or an
+   *   empty string if it fails. The method logs the outcome of the operation.
+   */
+  suspend fun addExpenseToTrip(tripId: String, expense: Expense): String =
```

Comment:
As discussed, Some people needed the object id immediatly after adding the operation, so now the return type is a string representing the ID of the new object added to firebase in case of succes, else empty string to notifiy an error.